modules/candidate_list/php/candidatelistrowprovisioner.class.inc

```diff
@@ -2,42 +2,49 @@
 
 namespace LORIS\candidate_list;
 
+use \LORIS\Data\DAO\Candidate_DAO;
+use \LORIS\Data\DTO\Candidate_DTO;
+use \LORIS\Data\DAO\Site_DAO;
+use \LORIS\Data\DTO\Site_DTO;
+
 class CandidateListRowProvisioner extends \LORIS\Data\Provisioner {
     protected function GetAllRows() : array {
-        $sql = "SELECT psc.Name AS Site,
-            c.CenterID,
-            c.CandID,
-            c.PSCID,
-            c.Gender,
-            c.Entity_type as `Entity Type`,
-            COALESCE(pso.Description,'Active') as `Participant Status`,
-            GROUP_CONCAT(DISTINCT sp.title) as Subproject,
-            DATE_FORMAT(c.DoB,'%Y-%m-%d') AS DoB,
-            MAX(s.Scan_done) as `Scan Done`,
-            COUNT(DISTINCT s.Visit_label) as `Visit Count`,
-            MAX(s.Current_stage) as `Latest Visit Status`,
-            IFNULL(MIN(feedback_bvl_thread.Status+0),0) as Feedback
-        FROM candidate c
-              LEFT JOIN psc ON (c.CenterID=psc.CenterID)
-              LEFT JOIN session s ON (c.CandID = s.CandID AND s.Active = 'Y')
-              LEFT JOIN feedback_bvl_thread
-                  ON (c.CandID=feedback_bvl_thread.CandID)
-              LEFT JOIN participant_status ps ON (ps.CandID=c.CandID)
-              LEFT JOIN participant_status_options pso
-                  ON (ps.participant_status=pso.ID)
-              LEFT JOIN subproject sp ON (s.SubprojectID=sp.SubprojectID)
-             WHERE c.Active = 'Y' 
-             GROUP BY c.CandID";
-        $db = \Database::singleton();
-        $query = $db->pselect($sql, array());
-        $results = [];
-
-        foreach ($query as $row) {
-            $cid = $row['CenterID'];
-            unset($row['CenterID']);
-            $r = new CandidateListRow($row, $cid);
-            $results[] = $r;
-        }
-       return $results; 
+        $candidate_dao = new Candidate_DAO();
+        $site_dao      = new Site_DAO();
```

Comment:
I don't have time to look at this in depth right now, but superficially I don't think these classes need "DAO" in the name since it's part of the namespace name it's redundant (and looks kind of ugly to have underscores in the class name)

I don't know if PHP allows it, but can you "use \LORIS\Data" and then refer to the classname as`DAO\Candidate`?

Reply:
Seems fare

Reply:
Same thing for DTO